tg_global_gsql.yaml

```diff
@@ -0,0 +1,21 @@
+---
+# @           : this file
+# {x}         : the directory of file x
+# &x          : any file which matches pattern x
+# ...         : any depth of arbitrary directories
+# *           : wildcard character
+# [x|y...|z]  : file/directory x, y, or z
+
+  algorithm_navigation:
+    category:
+      pattern: "{@}/algorithms/tg_category*.yml"
+    sub_category:
+      pattern: "{&category}/.../tg_sub*.yml"
+    algorithm:
+      pattern: "[{&category}|{&sub_category}/.../tg_algo*.yml"
```

Comment:
as we may need to support nested sub category(client use glob to get all sub category yaml file and use folder structure to recreate the sub category  hierarchy)
so algorithm pattern should change to `{&category}/.../tg_algo*.yml` , so client use folder structure to match algorithm to corresponding  sub category
```
pattern: "{&category}/.../tg_algo*.yml"
```